crypto/ctype.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <e_os.h>
+#include "internal/ctype.h"
+
+static const unsigned short ctype_char_map[128] = {
```

Comment:
I'd like to see a big comment here (or in internal/ctype.h) about how this is purposely using ASCII for ASN.1 operations, etc. So, in the future, someone doesn't use/modify these incorrectly.